gapic-generator-cloud/templates/cloud/service/rest/client/method/def/_rescue.erb

```diff
@@ -1,4 +1,7 @@
  <%- assert_locals method -%>
 rescue ::Faraday::Error => e
-  gapic_error = ::Gapic::Rest::Error.wrap_faraday_error e
-  raise ::Google::Cloud::Error.from_error(gapic_error)
+  begin
+    raise ::Gapic::Rest::Error.wrap_faraday_error e
+  rescue ::Gapic::Rest::Error => gapic_err
+    raise ::Google::Cloud::Error.from_error gapic_err.cause
+  end
```

Comment:
Am I reading this correctly? It seems like when the `Gapic::Rest::Error` is constructed, the cause is automatically set to `e`. And furthermore `wrap_faraday_error` _always_ returns a `Gapic::Rest::Error, so the rescue block here always executes, and then you throw away the `gapic_err` and just call `from_error(e)`. In other words, this entire section simplifies to:

```ruby
raise ::Google::Cloud::Error.from_error e
```

Reply:
I want the ::Gapic::Rest::Error to be the `cause` of ::Google::Cloud::Error

Reply:
Ohh, that's right, we were talking about that. Sorry, I missed that. Carry on. :)